ethereum/contracts/ShutdownSwitch.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Abstract contract module that provides a bridge contract with the ability to temporarily suspend transfers.
+ *
+ * Inheriting from `ShutdownSwitch` will allow a bridge contract to use the isEnabled modifier to prevent transactions
+ * when the shutdown switch is engaged via a guardian voting mechanism.
+ *
+ * This contract provides a public function that allows the guardians to vote to disable / enable transfers.
+ * When a predetermined number of guardians vote to disable, then transfers are blocked until the number of
+ * disable votes is reduced below that threshold.
+ *
+ * NOTE: The number of votes to enable is irrelevant. As long as there are enough votes to disable, trabsfers are blocked.
+ *
+ * An important design goal of this implementation is to minimize the impact (gas) to users, so the check to determine
+ * if transfers are enabled is a simple boolean check.
+ *
+ * The shutdown status is only updated when a vote is cast. When that happens, this contract gets the current guardian set.
+ * It validates that the voter (message sender) is an active guardian. If so, it updates their vote, counts the number of
+ * votes to disable, and updates the shutdown status.
+ *
+ * NOTE: The status is updated when any active guardian casts a vote, whether it changes anything or not. This means that,
+ * after a guardian set update, the shutdown status can be updated by casting an enable vote when you are already enabled.
+ *
+ * This contract provides public getters to query the shutdown status. It also emits events when valid votes are cast,
+ * or the shutdown status changes.
+ */
+
+import "./interfaces/IWormhole.sol";
+
+abstract contract ShutdownSwitch {
+    /// @dev A map of all guardians that have active votes to disable. If a guardian is removed from the guardian set while they
+    /// have an active vote to disable, they would get left in the map. The current assumption is that this will have minimal impact.
```

Comment:
Suggestion: "The current assumption is that this will have minimal impact because we will never reference that old guardian again in future passes"

Reply:
Added that detail